### PR TITLE
ExecutionProof: impl Serializable & Deserializable

### PR DESCRIPTION
**Pull request by hackaugusto**
## Describe your changes

Implements serializable and deserializable for execution proof (dependency to make ProvenTransaction serializable)

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,3 +22,11 @@ inherits = "release"
...
 debug-assertions = true
 overflow-checks = true
+
+[patch.crates-io]
+winter_air = { git = "https://github.com/facebook/winterfell", package = "winter-air" }
+winter_math = { git = "https://github.com/facebook/winterfell", package = "winter-math" }
+winter_crypto = { git = "https://github.com/facebook/winterfell", package = "winter-crypto" }
+winter_prover = { git = "https://github.com/facebook/winterfell", package = "winter-prover" }
+winter_verifier = { git = "https://github.com/facebook/winterfell", package = "winter-verifier" }
+winter_utils = { git = "https://github.com/facebook/winterfell", package = "winter-utils" }
```
Comment (bobbinth) [inline Cargo.toml:32]: The updated versions of winterfell crates are now on crates.io - so, this may not be necessary.

File: air/Cargo.toml [hunk 2]
```diff
@@ -30,7 +30,7 @@ std = ["vm-core/std", "winter-air/std"]
 
 [dependencies]
 vm-core = { package = "miden-core", path = "../core", version = "0.8", default-features = false }
-winter-air = { package = "winter-air", version = "0.7", default-features = false }
+winter-air = { package = "winter-air", git = "https://github.com/facebook/winterfell", default-features = false }
```
Comment (bobbinth) [inline air/Cargo.toml:33]: Similar comment to the one above.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I left a couple of minor comments inline. Once these are addressed, we can merge.

File: Cargo.toml (hunk 1 above)
Comment (hackaugusto) [reply Cargo.toml]: removed

File: air/Cargo.toml (hunk 2 above)
Comment (hackaugusto) [reply air/Cargo.toml]: changed back

**Pull request merged**